feat(provider/perplexity): add support for return_images

kevinwolfcr opened the pull request:
Adds support for [`return_images`](https://docs.perplexity.ai/api-reference/chat-completions#body-return-images) to the [Perplexity Provider](https://sdk.vercel.ai/providers/ai-sdk-providers/perplexity)

> [!NOTE]
> As per @shaper's recommendation, I'm _temporarily_ adding the logic to extract and pass the options to the underlying request in `openai-compatible-chat-language-model.ts`, since the Perplexity provider extends it.

shaper commented:
Hi there, thank you for the contribution! I just sent https://github.com/vercel/ai/pull/4618 which should allow us to do the input part of this PR as:

```
  const result = await generateText({
    model: perplexity('sonar-pro'),
    prompt: 'Invent a new holiday and describe its traditions.',
    providerOptions: {
      perplexity: {
        request_images: true,
      },
    },
  });
```

and similar for `streamText`. This should mean no changes needed for `openai-compatible`. I see you've kindly also done the extractor logic and tests, so I will work with you to get this landed once the other change is in (likely tomorrow).

Goal is to get you operational ASAP!

kevinwolfcr commented:
@shaper changes done. I removed the `returnImages` option from the `openai-compatible-chat-language-model.ts` and added it to the `providerOptions` object in the `perplexity-with-images` example for `streamText` and `generateText`.

shaper commented:
> @shaper changes done. I removed the `returnImages` option from the `openai-compatible-chat-language-model.ts` and added it to the `providerOptions` object in the `perplexity-with-images` example for `streamText` and `generateText`.

Thanks, I am looking at this

shaper commented:
Continuing this work (inclusive of your commits to track credit) in #4651
